parts/k8s/kubernetesagentcustomdata.yml

```diff
@@ -28,6 +28,15 @@ write_files:
   owner: root
   content: !!binary |
     {{WrapAsVariable "provisionConfigs"}}
+    
```

Comment:
Are we sure we need this custom controller manager scaffolding on agent nodes? For regular (non-Azure Stack) scenarios, the controller manager implementation is only on master nodes.

Reply:
It is also true that controller manager implementation is only on master nodes.
I updated ensureCertificates to check the existence kube controller file before update it.
The logic now are:
 

- on Master trust the certificate and update the kube controller config 

- on Agent, only trust the certificate